packages/input-text/src/TextInput.js

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import Tooltip from "@crave/farmblocks-tooltip";
+import Image, { badgeSizes } from "@crave/farmblocks-image";
+
+import errorIconSrc from "./constants/errorIcon";
+import StyledLabel from "./styledComponents/TextInput";
+
+class TextInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value,
+      invalid: this.props.invalid,
+      showTooltip: false
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onInvalid = this.onInvalid.bind(this);
+    this.onMouseOver = this.onMouseOver.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+  }
+
+  render() {
+    const {
+      label,
+      value,
+      onChange,
+      invalid,
+      validationErrors,
+      onInvalid,
+      tooltipText,
+      errorIconSrc,
+      onMouseOver,
+      onMouseLeave,
+      ...otherProps
+    } = this.props;
+    const containerProps = {
+      filled: this.state.value.length > 0,
+      invalid: this.state.invalid,
+      disabled: otherProps.disabled
+    };
+    const inputProps = {
+      value: this.state.value,
+      onChange: this.onChange,
+      onInvalid: this.onInvalid,
+      onMouseOver: this.onMouseOver,
+      onMouseLeave: this.onMouseLeave,
+      ...otherProps
+    };
+    return (
+      <StyledLabel {...containerProps}>
+        {this._renderInput(inputProps)}
+        {this._renderLabel(label)}
+        {this._renderTooltip(this.state.showTooltip, tooltipText)}
+        {this._renderFailedMessages(
+          this.state.invalid,
+          validationErrors,
+          errorIconSrc
+        )}
+      </StyledLabel>
+    );
+  }
+
+  _renderLabel(label) {
+    return label && <div className="label">{label}</div>;
+  }
+
+  _renderInput(inputProps) {
+    return <input type="text" {...inputProps} />;
+  }
+
+  _renderTooltip(visible, text) {
+    return (
+      visible && (
+        <div className="tooltip">
+          <Tooltip text={text} />
+        </div>
+      )
+    );
+  }
+
+  _renderFailedMessages(invalid, errors, iconSrc) {
+    return (
+      invalid && (
+        <div className="messages">
+          {errors.map((text, index) => (
+            <div className="message" key={`err-${index}`}>
+              <Image
+                className="icon"
+                src={iconSrc}
+                badge
+                size={badgeSizes.SMALL}
+              />
+              {text}
+            </div>
+          ))}
+        </div>
+      )
+    );
+  }
+  onChange(event) {
+    this.setState({
+      value: event.target.value,
+      invalid: false
```

Comment:
Not sure about this... This will work with the `required` prop and with other validations... (?)

Reply:
this version of the component dont have support for custom validations, and I don;t know if the DOM provides hooks to the native validations, I can investigate it. Yes, this hardcoded simple "have changed = valid now" is not ideal

Reply:
by other validations do you mean https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation ? 

I only tested with ``required`` (the "missing" validation) I haven't covered the other cases like type, pattern, underflow, overflow, step and too long

I will take a look at the link above and also this interface https://developer.mozilla.org/en-US/docs/Web/API/ValidityState and see what can be done to this milestone

Reply:
a better value for this invalid flag would be ``event.target.validity.valid`` that is the property when using the built-in validation

Reply:
Nice!